torch/_inductor/codegen/cpp.py

```diff
@@ -2372,9 +2373,10 @@ def reduction(self, dtype, src_dtype, reduction_type, value):
         if self.tiling_idx >= self.reduction_depth:
             # Horizontal reduction
             if is_welford_reduction(reduction_type):
-                assert (
-                    self._get_num_vectors(dtype) == 1
-                ), "Welford reduction does not support VectorizedN (N>1)"
+                assert self._get_num_vectors(dtype) in [
+                    1,
+                    2,
+                ], "Welford reduction does not support VectorizedN (N>2)"
```

Comment:
Perhaps it is not that hard to support arbitrary "N" if we already support N==2?

Reply:
It should be possible. In future PRs, we will support cases with N>=2 as needed.